src/Evolution/Systems/GeneralizedHarmonic/Equations.cpp

```diff
@@ -280,8 +280,10 @@ void ComputeDuDt<Dim>::apply(
   for (size_t n = 0; n < Dim; ++n) {
     for (size_t mu = 0; mu < Dim + 1; ++mu) {
       for (size_t nu = mu; nu < Dim + 1; ++nu) {
-        three_index_constraint.get(n, mu, nu) =
-            d_spacetime_metric.get(n, mu, nu) - phi.get(n, mu, nu);
+        const size_t storage_index =
+            three_index_constraint.get_storage_index(n, mu, nu);
+        three_index_constraint[storage_index] =
```

Comment:
The two-step indexing is a little cumbersome, especially if we're going to start peppering it in all of the performance-sensitive tensor calculations. Any chance we could make the run-time `get`s into a lookup into a compile-time table if we want the index computations to have the extra optimization?

Reply:
They already do this, and it's that index collapsing that can end up being a bit slow. However, I'll drop the commit from this PR and stick it on a branch. Once we are at a point where things work we can revisit this.

Reply:
alright, sounds good -- it would be nice to get the index access well-optimized, but I'm hopeful for a general solution so we don't have to do it manually everywhere. :)

Reply:
Agreed :) Full-blown tensor expressions should be able to do that since they would be able to generate the for loops, and hence the storage indices at compile time :)